cmd/buildkitd/config/config.go

```diff
@@ -42,9 +34,17 @@ type GRPCConfig struct {
 }
 
 type RegistryConfig struct {
-	Mirrors   []string `toml:"mirrors"`
-	PlainHTTP *bool    `toml:"http"`
-	Insecure  *bool    `toml:"insecure"`
+	Mirrors      []string     `toml:"mirrors"`
+	PlainHTTP    *bool        `toml:"http"`
+	Insecure     *bool        `toml:"insecure"`
+	RootCAs      []string     `toml:"ca"`
+	KeyPairs     []TLSKeyPair `toml:"keypair"`
```

Comment:
`toml:"keypairs"` ?

Reply:
All of them are arrays really 🤷‍♂

Reply:
Looking at the toml it actually makes more sense to keep it singular for complex types because they need to be defined one-by-one and plural for simple types. Already quite a mess though.